src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -63,7 +63,7 @@ private ViewDataDictionary CreateViewDataDictionary(ViewContext context, PageAct
             if (context.ViewData == null)
             {
                 // Create ViewDataDictionary<TModel>(IModelMetadataProvider, ModelStateDictionary).
-                return (ViewDataDictionary)Activator.CreateInstance(
+                return ViewDataDictionaryFactory.Create(
```

Comment:
Would using `ActivatorUtilities` with a null `IServiceProvider` work? Alternatively could we add an overload to it that supports it? Seems like we might be able to reuse a bunch of code if we do that.